Reference types missing from tensor.d.ts

<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
Master tfjs branch on any system.

**Describe the current behavior**
Reference types for `@webgpu/types` are missing when compiling `tfjs-core/src/tensor.ts`. These types are necessary because #6329 added the `GPUBuffer` type to `tensor.ts`, and their absence causes downstream packages to fail to compile (#6329 passed presubmits and was merged because of #6384).

**Describe the expected behavior**

`tensor.d.ts` contains the following triple slash reference types at the top of the file:
```typescript
/// <reference types="@webgpu/types" />
```

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/CodePen/any notebook.

`yarn bazel build //tfjs-core:tfjs-core_pkg` and then check `dist/bin/tfjs-core/tfjs-core_pkg/dist/tensor.d.ts`

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

```
tfjs-backend-wasm$ yarn build
$ rimraf dist/ && tsc && ./scripts/build-wasm.sh
../link-package-core/node_modules/@tensorflow/tfjs-core/dist/tensor.d.ts:77:14 - error TS2304: Cannot find name 'GPUBuffer'.

77     buffer?: GPUBuffer;
                ~~~~~~~~~


Found 1 error.

error Command failed with exit code 2.
```

--- a/tfjs-core/package.json
+++ b/tfjs-core/package.json
@@ -76,6 +76,7 @@
     "@types/offscreencanvas": "~2019.3.0",
     "@types/seedrandom": "2.4.27",
     "@types/webgl-ext": "0.0.30",
+    "@webgpu/types": "^0.1.16",
     "long": "4.0.0",
     "node-fetch": "~2.6.1",
     "seedrandom": "2.4.3"

--- a/tfjs-core/src/tensor.ts
+++ b/tfjs-core/src/tensor.ts
@@ -15,6 +15,9 @@
  * =============================================================================
  */
 
+// Workaround for: https://github.com/bazelbuild/rules_nodejs/issues/1265
+/// <reference types="@webgpu/types/dist" />
+
 import {getGlobal} from './global_util';
 import {tensorToString} from './tensor_format';
 import {ArrayMap, BackendValues, DataType, DataTypeMap, DataValues, NumericDataType, Rank, ShapeMap, SingleValueMap, TypedArray} from './types';

--- a/tfjs-core/tsconfig.json
+++ b/tfjs-core/tsconfig.json
@@ -7,9 +7,6 @@
     "node_modules/"
   ],
   "compilerOptions": {
-    "typeRoots": [
-      "../node_modules/@webgpu"
-    ],
     "outDir": "./dist"
   }
 }

--- a/tsconfig_ts_library.json
+++ b/tsconfig_ts_library.json
@@ -1,6 +1,5 @@
 {
   "compilerOptions": {
-    "typeRoots": [ "./node_modules/@webgpu", "./node_modules/@types"],
     "module": "ES2015",
     "moduleResolution": "node",
     "noImplicitAny": true,

